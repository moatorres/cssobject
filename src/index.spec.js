import toObj from './index'

describe('toObj', () => {
  it('Should be defined', () => {
    expect(toObj).toBeDefined()
  })

  it('Should allow us to convert CSS string-literals to JavaScript objects', () => {
    const options = {
      camelCase: false,
      kebabCase: false,
      numbers: false,
    }

    const style = toObj(
      `.button {
        color: tomato;
        margin-bottom: 16px;
        @media (min-width: 40em) {
          color: green;
        }
      }`,
      options
    )

    expect(typeof style).toEqual('object')
    expect(style['.button']).toBeDefined()
    expect(style['.button']).toEqual({
      color: 'tomato',
      'margin-bottom': 16,
    })
  })

  it('Should correctly parse pseudo-classes', () => {
    const style = toObj(`
    .button {
      color: tomato;
      &:hover {
        color: green;
      }
    }
    `)

    expect(style).toMatchSnapshot()
  })

  it('Should match the snapshot', () => {
    const style = toObj(
      `.button {
          color: tomato;
          &:hover {
            color: black;
            fontSize: 1rem;
          }
        }`,
      {
        camelCase: true,
        numbers: true,
      }
    )

    expect(style).toMatchSnapshot()
  })
})
