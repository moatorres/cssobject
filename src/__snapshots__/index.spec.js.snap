// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toObj Should correctly parse pseudo-classes 1`] = `
Object {
  ".button": Object {
    "color": "tomato",
  },
  ".button:hover": Object {
    "color": "green",
  },
}
`;

exports[`toObj Should match the snapshot 1`] = `
Object {
  ".button": Object {
    "color": "tomato",
  },
  ".button:hover": Object {
    "color": "black",
    "fontSize": "1rem",
  },
}
`;
